processor/src/host/mod.rs

```diff
@@ -196,3 +196,56 @@ impl Host for DefaultHost {
         Ok(())
     }
 }
+
+// ASYNC HOST trait
+// ================================================================================================
+
+/// Analogous to the [Host] trait, but designed for asynchronous execution contexts.
+pub trait AsyncHost {
+    // REQUIRED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a reference to the advice provider.
+    fn advice_provider(&self) -> &AdviceProvider;
+
+    /// Returns a mutable reference to the advice provider.
+    fn advice_provider_mut(&mut self) -> &mut AdviceProvider;
+
+    /// Returns MAST forest corresponding to the specified digest, or None if the MAST forest for
+    /// this digest could not be found in this [Host].
+    fn get_mast_forest(
+        &mut self,
+        node_digest: &Word,
+    ) -> impl Future<Output = Option<Arc<MastForest>>> + Send;
```

Comment:
Question: does using native `async` here complicate things? i.e.:
```rust
async fn get_mast_forest(
    &mut self,
    node_digest: &Word,
) -> Option<Arc<MastForest>>>
```

Reply:
Makes sense! I'd add a comment somewhere briefly explaining this.

Also, not sure if it is relevant, but came across this repo: https://github.com/spastorino/dynosaur which has something to do with using dynamic dispatch with async traits.

Reply:
I don't think we'll need it. host are usually short-lived and we also usually use concrete host implementations (that's how the transaction executor and host work). So, probably we can ignore the `dynasaur` for now.

Reply:
> Makes sense! I'd add a comment somewhere briefly explaining this.

Done

Reply:
Heads up, but I think if we move our MSRV to the latest stable, we'll pick up access to stablized async-fn-in-traits. That blog post was released before that feature was stabilized, and the reason we get a warning currently is because we're on an MSRV that predates the stabilization.